base/stats.go

```diff
@@ -0,0 +1,747 @@
+package base
+
+import (
+	"expvar"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	StatViewFormat = "%s.%s"
+)
+
+type SgwStats struct {
+	GlobalStats     GlobalStat          `json:"global"`
+	DbStats         map[string]*DbStats `json:"per_db"`
+	ReplicatorStats *ReplicatorStats    `json:"per_replication"`
+
+	initDBStats               sync.Once
+	registeredReplicatorStats sync.Once
+}
+
+func (s *SgwStats) String() string {
+	bytes, _ := JSONMarshal(s)
```

Comment:
I think this needs to be JSONMarshalCanonical, to provide the key ordering that's expected for expvar maps.